src/Plugins/Reminder.php

```diff
@@ -0,0 +1,357 @@
+                case 'reminder': case 'remind': break;
+                case 'in': case 'at': return yield $this->setReminder($command); break;
+            }
+
+            if ( count(array_diff($command->getParameters(), array($textOrCommand))) < 1 ){
+                switch ($textOrCommand){
+                    case 'list': return yield $this->getAllReminders($command); break;
+                    case 'examples': return yield $this->getExamples($command); break;
+                    case 'flush': return yield $this->flush($command);
+                }
+            }
+
+            if( $command->getParameter(0) === 'unset'
+                && $command->getParameter(1) !== null
+                && count($command->getParameters()) <= 2
+            ){ return yield $this->unset($command); }
+
+            return yield $this->setReminder($command);
+        });
+    }
+
+    public function apologize(ChatRoom $room, $timedOutReminders): \Generator
+    {
+        if(count($timedOutReminders) > 0){
+            foreach ($timedOutReminders as $key){
+                $value = yield $this->storage->get($key,$room);
+                $text = $value['text'];
+                $name = $value['username'];
+
+                if (null !== $pingableName = yield $this->chatClient->getPingableName($room, $name)) {
+                    $target = "@{$pingableName}";
+                    $reply = $target . " I guess I'm late: " . $text;
+                    yield $this->storage->unset($key, $room);
+                    return $this->chatClient->postMessage($room, $reply, PostFlags::ALLOW_PINGS);
+                }
+            }
+        }
+    }
+
+    public function reschedule(ChatRoom $room, array $upcomingReminders): \Generator
+    {
+        if(count($upcomingReminders) > 0){
+            $this->watchers = [];
+
+            foreach ($upcomingReminders as $key){
+                $value = yield $this->storage->get($key,$room);
+                $text  = $value['text'];
+                $name  = $value['username'];
+                $stamp = $value['timestamp'];
+                $seconds = $stamp - time();
+
+                if (null !== $pingableName = yield $this->chatClient->getPingableName($room, $name)) {
+                    $target = "@{$pingableName}";
+                    $reply = ($seconds <= 0) ? $target ." I guess I'm late: " . $text : $target . " " . $text;
+
+                    if ($seconds <= 0): return $this->chatClient->postMessage($room, $reply); endif;
+
+                    $this->watchers[] = once(function () use ($room, $key, $value, $reply) {
+                        yield $this->storage->unset($key, $room);
+                        return $this->chatClient->postMessage($room, $reply);
+                    }, $seconds * 1000);
+                }
+            }
+        }
+    }
+
+    public function enableForRoom(ChatRoom $room, bool $persist = true){
+        $timedOutReminders = $upcomingReminders = [];
```

Comment:
No need for these anymore because we are just going to pass all reminders to the methods.